Framework/Built_In_Automation/Sequential_Actions/common_functions.py

```diff
@@ -7107,3 +7107,130 @@ def download_chrome_extension(data_set):
+
+    Args:
+        data_set:
+            ------------------------------------------------------------------------------
+            image                       | input parameter | %| image.png |%
+            user prompt                 | optional parameter | Extract invoice details
+            AI visual reader            | common action   | AI visual reader
+            ------------------------------------------------------------------------------
+    
+    Return:
+        `passed` if success
+        `zeuz_failed` if fails
+    """
+    sModuleInfo = inspect.currentframe().f_code.co_name + " : " + MODULE_NAME
+    global selenium_driver
+
+    try:
+        import base64
+        import requests
+        import json
+        import os
+        user_image_path = None
+        user_prompt = None
+        
+        for left, mid, right in data_set:
+            left = left.lower().replace(" ", "")
+            mid = mid.lower().replace(" ", "")
+            right = right.strip()
+
+            if left == 'image':
+                if right != '':
+                    user_image_path = right
+            
+            if left == "userprompt":
+                if right != '':
+                    user_prompt = right
+        
+        # Validate image path
+        if not user_image_path:
+            CommonUtil.ExecLog(sModuleInfo, "No image path provided. Please provide an image path.", 3)
+            return "zeuz_failed"
+        
+        image_path = user_image_path
+        CommonUtil.ExecLog(sModuleInfo, f"Processing image: {image_path}", 1)
+        
+        if not os.path.isfile(image_path):
+            CommonUtil.ExecLog(sModuleInfo, f"Image file not found: {image_path}", 3)
+            return "zeuz_failed"
+
+        prompt = user_prompt
+        if not prompt:
+            prompt = "Extract all text from this image and return the result in JSON format."
+
+        # Convert Image to Base64
+        with open(image_path, "rb") as img_file:
+            base64_image = base64.b64encode(img_file.read()).decode("utf-8")
+
+        # Load API key from .env file
+        try:
+            from dotenv import load_dotenv
+            framework_dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+            env_path = os.path.join(framework_dir, ".env")
+            load_dotenv(env_path)
+            api_key = os.getenv("OPENAI_API")
+            if not api_key:
+                CommonUtil.ExecLog(sModuleInfo, "OPENAI_API not found in .env file", 3)
+                return "zeuz_failed"
+        except Exception as e:
+            CommonUtil.ExecLog(sModuleInfo, f"Failed to load API key from .env: {str(e)}", 3)
+            return "zeuz_failed"
+
+        # Prepare API Request
+        headers = {
+            "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json"
+        }
+
+        payload = {
+            "model": "gpt-4o",
```

Comment:
Users should be able to specify the model name as part of the action parameters.
They should also be able to configure the temperature and any other associated hyperparameters of the model.